### ISSUE TITLE
task model: not correctly checking (inferring?) initializing formals

### ISSUE THREAD

**Issue by jmesserly** [dart-lang/dev_compiler#346]
This test is failing:
```
00:03 +53 -5: inferred initializing formal checks default value
  expected error was not reported at:
  
  severe: line 3, column 49 of /main.dart: StaticTypeError
          Foo([this.x = /*severe:StaticTypeError*/"1"]);
                                                  ^^^
```

**Comment by jmesserly** [comment]
I believe this also affects 'Constructors' in checker_test.dart

**Comment by jmesserly** [comment]
the "Constructors" test is now passing with @leafpetersen 's CL https://codereview.chromium.org/1370793002/

**Comment by jmesserly** [comment]
the other one seems to be failing still

**Comment by jmesserly** [comment]
Ah, here's what our old code had to say:
```dart
  @override
  visitFieldFormalParameter(FieldFormalParameter node) {
    // Ensure the field formal parameter's type is updated after inference.
    // Normally this happens during TypeResolver, but that's before we've done
    // inference on the field type.
    var element = node.element;
    if (element is FieldFormalParameterElement) {
      if (element.type.isDynamic) {
        // In malformed code, there may be no actual field.
        if (element.field != null) {
          element.type = element.field.type;
        }
      }
    }
    super.visitFieldFormalParameter(node);
  }
```

I wonder if we have the same issue ... that we're inferring after the type has been computed.

**Comment by jmesserly** [comment]
fixed here: https://codereview.chromium.org/1371813002/
